edx/edx-app-ios
# iOS - Integrate Firebase Remote Config

### Description

[LEARNER-7999](https://openedx.atlassian.net/browse/LEARNER-7999)

This PR has integrated Firebase remote config. 

### How to test this PR
- I have added a new parameter `app_theme` with json 
`app_theme: {
  font: {
    name: "fonts-file-name",
    enabled: false
  },
  color: {
    name: "colors-file-name",
    enabled: false
  },
  icon: "new-icon-name",
  mode: "dark/light/system"
}
`
- To test the remote config we can update the `fetchRemoteConfig` method in AppDelegate class and uncomment to have remoteConfig object then we can see the different values by printing.
```
- (void)fetchRemoteConfig {
    [self.firebaseRemoteConfig fetchWithCompletionHandler:^(FIRRemoteConfigFetchStatus status, NSError *error) {
        if (status == FIRRemoteConfigFetchStatusSuccess) {
            [self.firebaseRemoteConfig activateWithCompletionHandler:^(NSError * _Nullable error) {
                
            }];
            
            //Uncomment this when to use remoteConfig
            //FirebaseRemoteConfiguration* remoteConfig = [[FirebaseRemoteConfiguration alloc] initWithRemoteConfig:self.firebaseRemoteConfig];
            
        } else {
            NSLog(@"Error %@", error.localizedDescription);
        }
    }];
}
```
